### ISSUE TITLE
Minor problems & potential enhancements related to menu buttons & windows

### ISSUE THREAD

**Issue by tomchen** [arcomage/arcomage-hd#55]
- [x] Menu buttons should be clickable with middle and right mouse buttons.
- [x] Click a menu button, it shows a preference window, click the same menu button, it should close the window, not close and quickly show it again.
- [x] When a preference window is open, <kbd>Esc</kbd> is supposed to close the window, but in fullscreen mode, <kbd>Esc</kbd> will exit the fullscreen mode and not close the window.
- [x] Menu buttons shouldn't have context menu
